src/device_impl.rs

```diff
@@ -118,4 +123,16 @@ where
         };
         self.iface.write_register(Register::CMD, cmd)
     }
+
+    /// Set the accelerometer range
+    pub fn set_accel_range(&mut self, range: AccelerometerRange) -> Result<(), Error<CommE>> {
+        self.accel_range = range;
+        self.iface.write_register(Register::ACC_RANGE, range as u8)
```

Comment:
```suggestion
        let res = self.iface.write_register(Register::ACC_RANGE, range as u8)?;
        self.accel_range = range;
        res
```
This way we avoid storing something if the write fails.